src/core/src/dimension.cpp

```diff
@@ -41,6 +41,20 @@ Dimension Dimension::operator-(const Dimension& dim) const {
     return Dimension(m_dimension - dim.m_dimension);
 }
 
+Dimension Dimension::operator/(const value_type divisor) const {
+    OPENVINO_ASSERT(divisor >= 0, "divisor must be greater than 0");
+
+    if (m_dimension.get_max_val() == Interval::s_max && m_dimension.get_min_val() == 0)
+        return Dimension::dynamic();
+
+    else if (is_static() && m_dimension.get_max_val() % divisor)
+        OPENVINO_UNREACHABLE("Dimension value: ",
```

Comment:
```suggestion
Dimension Dimension::operator/(const value_type divisor) const {
    OPENVINO_ASSERT(divisor >= 0, "divisor must be greater than 0");

    if (m_dimension.get_max_val() == Interval::s_max && m_dimension.get_min_val() == 0)
        return Dimension::dynamic();

   if (is_static() && m_dimension.get_length() % divisor)
        OPENVINO_UNREACHABLE("Dimension value: ",
                             m_dimension.get_max_val(),
                             " must be a multiple of divisor: ",
                             divisor);
       if ((m_dimension.get_max_val() / divisor  ==  m_dimension.get_min_val() / divisor) &&
            m_dimension.get_min_val() > (m_dimension.get_min_val() / divisor * divisor) )
        OPENVINO_UNREACHABLE("Dimension value: 
                              must be a multiple of divisor: ",
                             divisor);
    return Dimension(m_dimension.get_min_val() / divisor, m_dimension.get_max_val() / divisor)
}
```
I think this assert only check when dimension is static. When static and dimension value is not a multiple of divisor， the output should be empty. I switched to invoke Dimension.get_length() to avoid confusing.

Also the comments reminds me about the checking whether output is empty when dimension is dynamic. Update in the suggestion code. Let me know whether this makes sense to you. Thx!